integration/appaccess/appaccess_test.go

```diff
@@ -557,6 +557,56 @@ func TestTCP(t *testing.T) {
 	}
 }
 
+// TestTCPLock tests locking TCP applications.
+func TestTCPLock(t *testing.T) {
```

Comment:
Can you also add a test to cover "disconnect expired cert" scenario in addition to explicit locking?